pkg/cvo/availableupdates.go

```diff
@@ -159,13 +169,35 @@ func calculateAvailableUpdatesStatus(clusterID, upstream, channel, version strin
 	}
 }
 
-func checkForUpdate(clusterID, upstream, channel string, currentVersion semver.Version) ([]cincinnati.Update, error) {
+func checkForUpdate(clusterID, upstream, channel string, currentVersion semver.Version, url *url.URL) ([]cincinnati.Update, error) {
 	uuid, err := uuid.Parse(string(clusterID))
 	if err != nil {
 		return nil, err
 	}
+
 	if len(upstream) == 0 {
 		return nil, fmt.Errorf("no upstream URL set for cluster version")
 	}
-	return cincinnati.NewClient(uuid).GetUpdates(upstream, channel, currentVersion)
+	return cincinnati.NewClient(uuid, url).GetUpdates(upstream, channel, currentVersion)
+}
+
+func (optr *Operator) getHTTPSProxyURL() (*url.URL, error) {
+	proxy, err := optr.proxyLister.Get("cluster")
+
+	if errors.IsNotFound(err) {
```

Comment:
see https://github.com/openshift/cluster-version-operator/pull/219/files#r301772882

we should do a no-op when the type doesn't exist.